### ISSUE TITLE
PDF golden and cross-subcommittee tests skip in CI instead of running, and green hides it

### ISSUE THREAD

**Issue by willhea** [civictechdc/DeltaTrack#287]
**What's wrong**

A large block of the PDF structure tests never runs in CI, and CI reports green anyway.
They are written to skip themselves when their input file is missing, the input files are
not committed to the repo, and no CI step fetches them. So the tests are present, they
are collected, they report as skipped, and nothing distinguishes "this passed" from "this
never ran."

This is the same defect as #167 (corpus property gates skip silently when their asset
isn't fetched), which was fixed for the corpus property gates by #217 (run corpus gates
in CI from a committed manifest, fail-closed) and #220 (migrate the remaining corpus
modules off `REQUIRE_CORPUS`). The PDF anchor and extraction goldens were not part of
that migration and still have the original behavior.

**How it surfaced**

Noticed while adding a fail-closed fixture check in #284 (committed sample and example
reports drift behind the renderer). The pattern was deliberately not copied from
`tests/test_pdf_anchor_golden.py`, which prompted checking what that file actually does
in CI.

From the CI run on PR #286, "Run tests" step (`pytest -m "not slow and not browser"`),
which finished **1192 passed, 60 skipped**:

- **42 of those 60 skips are in `tests/test_pdf_anchor_golden.py`.** Another 5 are in
  `tests/test_pdf_extraction_golden.py`.
- The golden snapshot comparison runs for two of its three fixture bills and silently
  does not run for the third:

```
PASSED  tests/test_pdf_anchor_golden.py::test_anchors_match_golden[118-hr-8282]
PASSED  tests/test_pdf_anchor_golden.py::test_anchors_match_golden[118-hr-8752]
SKIPPED tests/test_pdf_anchor_golden.py::test_anchors_match_golden[118-s-4795]
```

- The entire cross-subcommittee validation set skips, including every case of
  `TestMajorLevelAcrossSubcommittees::test_signature_major_recovered_intact` and
  `test_stacked_headers_split`. That suite exists specifically to catch overfitting the
  heading detection to one or two bills, which is the thing least likely to be noticed by
  any other test.

The cause is fixture availability, not test logic:

```
$ git ls-files bills/118-hr-8752/1_reported-in-house.pdf   # 1  (tracked, so CI has it -> PASSED)
$ git ls-files bills/118-hr-8282/1_introduced-in-house.pdf # 1  (tracked -> PASSED)
$ git ls-files test_data/BILLS-118s4795rs.pdf              # 0  (not tracked -> SKIPPED)
$ git ls-files test_data/subcommittee/                     # 0  (not tracked -> all SKIPPED)
$ grep -c fetch_test_assets .github/workflows/ci.yml       # 0  (nothing fetches them)
```

`scripts/fetch_test_assets.py` downloads these from govinfo for local use. CI never runs
it, so on a CI checkout the files cannot be there.

**Why it matters**

The consequence is not that coverage is lower than believed. It is that the coverage
number gives no signal at all for these cases: a change that breaks heading detection on
eleven of the twelve appropriations subcommittees passes CI, because the tests that would
catch it skip rather than fail.

The affected tests are the ones guarding against overfitting. Their whole purpose is to
prove that heading and anchor detection generalizes past the two bills that happen to be
committed, and CI currently validates only those two bills. A green run is being read as
evidence of something it does not check.

It also fails in the direction that hides itself. Adding a fixture-dependent test to this
file today produces a test that appears to pass in CI while never executing, and the
author gets no warning.

**What to do**

The precedent from #217 and #220 is a committed fixture set plus a floor test that fails
closed when a fixture it expects is absent. Applying the same shape here has a cost
question attached, so the options differ mainly in what gets committed.

1. **Floor test only, no new fixtures.** Add a test asserting the expected fixture set is
   present, so an absent fixture goes red rather than silently disabling a test. Cheapest,
   and it makes the gap visible immediately. On its own it would turn CI red until the
   fixtures are dealt with, so it pairs with one of the options below rather than
   standing alone.
2. **Commit the fixtures, like the corpus manifest did.** Makes CI actually run these
   tests. Cost is repository size: the fixtures are PDF prints, and there are eleven of
   them (ten subcommittee prints plus `BILLS-118s4795rs.pdf`). This interacts directly
   with #126 (curate the CI test corpus: smaller, broader, deliberate coverage), which is
   open and is the right place to decide how much fixture weight the repo should carry.
3. **Fetch in CI.** Add a `scripts/fetch_test_assets.py` step to the workflow. No
   repository growth, but it makes CI depend on govinfo being reachable, which trades a
   silent skip for an intermittent failure and a slower run.
4. **Narrow the fixture set deliberately.** If twelve subcommittees is more than the
   invariant needs, pick a defensible subset, commit those, and record why in the test,
   so the coverage claim matches what runs. This is a #126 decision.

Options 2 and 4 need a size measurement that this investigation could not take, see below.

**Verification**

"The suite passes" cannot verify a fix here, because it passes now, which is the problem.
The check is that the previously-skipped tests appear as **passed** in the CI log, and
that removing a fixture turns the floor test red. Concretely: after the change, the CI
log for the "Run tests" step should show `test_anchors_match_golden[118-s-4795]` as
PASSED rather than SKIPPED, and the skip count for `tests/test_pdf_anchor_golden.py`
should be zero rather than 42.

**What was not checked**

The total size of the missing fixtures was not measured. `test_data/subcommittee/` does
not exist in either local checkout used for this investigation, so the ten subcommittee
PDFs could not be weighed. The one fixture that is present locally,
`test_data/BILLS-118s4795rs.pdf`, is 334 KB, but that is a single data point and should
not be multiplied out to estimate the rest. Measuring the set is a prerequisite for
choosing between options 2, 3 and 4.

Whether other test files share this pattern was also not audited beyond the two golden
modules named above. The skip counts show `tests/test_pdf_diff_recall.py` (5),
`tests/test_pdf_division_recall.py` (4) and `tests/test_pdf_text.py` (2) also skipping in
the same CI step, and some of those skips are legitimate content-based skips rather than
missing-fixture skips. Separating the two was out of scope here.

Refs #167, #217, #220, #126, #284
